### PR TITLE
feat: Simplify session handling

### PR DESCRIPTION
**Pull request by kaladivo**
## Summary by CodeRabbit

* **New Features**
  * Improved sign-in resilience when migrating from older versions by backfilling missing v2 secrets from legacy storage.

* **Bug Fixes**
  * Fixed session loading recovery when the app gets stuck in a stale “loading” state.
  * Improved concurrent session loads so multiple requests share a single in-progress storage read.
  * Added safeguards for hung storage reads with a watchdog timeout to enable continued recovery.

* **Tests**
  * Expanded coverage for watchdog timeout behavior, concurrency edge cases, and legacy-to-v2 secret migration.

### CODE REVIEW

**Review by SamTremko — APPROVED** [review]

File: apps/mobile/src/state/session/loadSession.ts [hunk 1]
```diff
@@ -208,62 +135,33 @@ function isBlockingRecoveryError(loadingError: LoadSessionError): boolean {
...
+            'warn',
+            new Error('Waiting for session load to finish timed out', {
+              cause: e,
+            })
+          ),
+          Effect.succeed(
+            sessionNotLoadedResult(new SessionLoadWaitTimedOut({cause: e}))
+          )
         )
       )
```
[the remark this answers is not in the document]
Comment (kaladivo) [reply apps/mobile/src/state/session/loadSession.ts:167]: Fixed in e88b1014b. The published in-flight promise is now the load raced against a 30s watchdog timer. When the watchdog fires, all waiting callers settle with a blocking-recovery `SessionLoadWaitTimedOut` result, the in-flight slot is freed and the session atom reset from 'loading' to 'initial', so the next call starts a fresh load. The hung load is deliberately not interrupted (avoids tearing a half-finished storage write) — if it ever finishes it still settles the session atom, and the slot ownership check keeps it from touching a newer load's slot. Covered by a new test.

File: apps/mobile/src/state/session/loadSession.ts [hunk 2]
```diff
@@ -449,19 +317,35 @@ export function loadSession(
...
+    const loadPromise = Effect.runPromise(performSessionLoad())
+    inFlightLoad = loadPromise
+    store.set(sessionHolderAtom, {state: 'loading'})
+    void loadPromise.finally(() => {
+      // Only the load that owns the slot may free it.
+      if (inFlightLoad === loadPromise) inFlightLoad = null
+    })
+
+    return Effect.promise(() => loadPromise)
   })
```
[the remark this answers is not in the document]
Comment (kaladivo) [reply apps/mobile/src/state/session/loadSession.ts:392]: Fixed in e88b1014b. The initiating caller now awaits the same published promise as joiners: the load raced against a 30s watchdog. The watchdog bound is intentionally longer than the 5s join cap — a legitimate v1→v2 session upgrade makes network calls and can take longer than 5s, but the initiator can no longer block forever: on watchdog timeout it gets a blocking-recovery result and the in-flight slot is freed for a fresh load.
